File: src/model/Services.ts

export class Services {
  private _idService !: number;
  private _ownerEmail !: string;
  private _name !: string;
  private _description !: string ;
  private _score !: number;
  private _approved !: boolean;
  private _datePublication !: Date;
  private _prix !: number ;
  private _durre !: string;
  private _hidden !: boolean;


  constructor(idService: number, ownerEmail: string, name: string, description: string, score: number, approved: boolean, datePublication: Date, prix: number, durre: string, hidden: boolean) {
    this._idService = idService;
    this._ownerEmail = ownerEmail;
    this._name = name;
    this._description = description;
    this._score = score;
    this._approved = approved;
    this._datePublication = datePublication;
    this._prix = prix;
    this._durre = durre;
    this._hidden = hidden;
  }


  get idService(): number {
    return this._idService;
  }

  set idService(value: number) {
    this._idService = value;
  }

  get ownerEmail(): string {
    return this._ownerEmail;
  }

  set ownerEmail(value: string) {
    this._ownerEmail = value;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get description(): string {
    return this._description;
  }

  set description(value: string) {
    this._description = value;
  }

  get score(): number {
    return this._score;
  }

  set score(value: number) {
    this._score = value;
  }

  get approved(): boolean {
    return this._approved;
  }

  set approved(value: boolean) {
    this._approved = value;
  }

  get datePublication(): Date {
    return this._datePublication;
  }

  set datePublication(value: Date) {
    this._datePublication = value;
  }

  get prix(): number {
    return this._prix;
  }

  set prix(value: number) {
    this._prix = value;
  }

  get durre(): string {
    return this._durre;
  }

  set durre(value: string) {
    this._durre = value;
  }

  get hidden(): boolean {
    return this._hidden;
  }

  set hidden(value: boolean) {
    this._hidden = value;
  }
}

File: lancersBackEnd/model/Services.py
from datetime import datetime

class Services:
    def __init__(self, idService: int, ownerEmail: str, name: str, description: str, score: int, approved: bool, datePublication: datetime, prix: float, durre: str, hidden: bool):
        self._idService = idService
        self._ownerEmail = ownerEmail
        self._name = name
        self._description = description
        self._score = score
        self._approved = approved
        self._datePublication = datePublication
        self._prix = prix
        self._durre = durre
        self._hidden = hidden

    @property
    def idService(self) -> int:
        return self._idService

    @idService.setter
    def idService(self, value: int):
        self._idService = value

    @property
    def ownerEmail(self) -> str:
        return self._ownerEmail

    @ownerEmail.setter
    def ownerEmail(self, value: str):
        self._ownerEmail = value

    @property
    def name(self) -> str:
        return self._name

    @name.setter
    def name(self, value: str):
        self._name = value

    @property
    def description(self) -> str:
        return self._description

    @description.setter
    def description(self, value: str):
        self._description = value

    @property
    def score(self) -> int:
        return self._score

    @score.setter
    def score(self, value: int):
        self._score = value

    @property
    def approved(self) -> bool:
        return self._approved

    @approved.setter
    def approved(self, value: bool):
        self._approved = value

    @property
    def datePublication(self) -> datetime:
        return self._datePublication

    @datePublication.setter
    def datePublication(self, value: datetime):
        self._datePublication = value

    @property
    def prix(self) -> float:
        return self._prix

    @prix.setter
    def prix(self, value: float):
        self._prix = value

    @property
    def durre(self) -> str:
        return self._durre

    @durre.setter
    def durre(self, value: str):
        self._durre = value

    @property
    def hidden(self) -> bool:
        return self._hidden

    @hidden.setter
    def hidden(self, value: bool):
        self._hidden = value

